gateway/src/main/java/io/zeebe/gateway/impl/broker/cluster/BrokerTopologyManagerImpl.java

```diff
@@ -79,123 +72,112 @@ public BrokerClusterState getTopology() {
   }
 
   @Override
-  public ActorFuture<BrokerClusterState> requestTopology() {
-    final CompletableActorFuture<BrokerClusterState> future = new CompletableActorFuture<>();
-
-    actor.run(
+  public void event(ClusterMembershipEvent event) {
+    actor.call(
         () -> {
-          final boolean isFirstStagedRequest = nextTopologyFutures.isEmpty();
-          nextTopologyFutures.add(future);
-
-          if (isFirstStagedRequest) {
-            scheduleNextRefresh();
-          }
+          LOG.info("Gateway: Received event: {}", event);
+          onMembershipEvent(event.type(), event.subject());
         });
-
-    return future;
   }
 
-  @Override
-  public void withTopology(Consumer<BrokerClusterState> topologyConsumer) {
-    final BrokerClusterStateImpl brokerClusterState = topology.get();
-    if (brokerClusterState != null) {
-      topologyConsumer.accept(brokerClusterState);
-    } else {
-      actor.run(
-          () ->
-              actor.runOnCompletion(
-                  requestTopology(),
-                  (topology, error) -> {
-                    if (error == null) {
-                      topologyConsumer.accept(topology);
-                    } else {
-                      withTopology(topologyConsumer);
-                    }
-                  }));
-    }
-  }
+  private void onMembershipEvent(Type eventType, Member subject) {
+    final Properties properties = subject.properties();
+    final String memberId = subject.id().id();
+    final int brokerId;
 
-  private void scheduleNextRefresh() {
-    final long now = ActorClock.currentTimeMillis();
-    final long timeSinceLastRefresh = now - lastRefreshTime;
+    try {
+      brokerId = Integer.parseInt(memberId);
+    } catch (NumberFormatException e) {
+      LOG.info("Gateway: Ignoring member named '{}'", memberId);
```

Comment:
I don't like this approach of having exceptions as part of the expected control flow. We should make sure that our member ids are normally parseable, instead of relying on this hack.

Reply:
follow up issue #1958